### PR TITLE
fix(header): add hide on scroll functionality

### LINKED ISSUE #58: [IMPROVEMENT] Hide header on scroll
The header should be hidden as we scroll down after reaching a threshold. And as we scroll up, it should be visible. 

Attached below is a codepen with the code in HTML/CSS/JS. Implement the same logic on top of React. Make sure to properly handle event listeners as not doing so would leak memory.

Codepen: https://codepen.io/iabhinavr/pen/BaKxvRx

### PR DESCRIPTION
**Pull request by OMIIIEE**
fix #58  fix hide header on scroll and ensured Cleanup event listener on component unmount to prevent any memory leak . 

https://github.com/user-attachments/assets/3e433882-37f7-494a-96ee-091c6d0e1049

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -21,6 +21,6 @@
       "~/*": ["./src/*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "src/hooks/useHeaderScroll.js"],
```
Comment (realChakrawarti) [inline tsconfig.json:24]: I don't see any `useHeaderScroll` hook. Is this development leftover? If so, could you please remove the change. Thanks.

File: src/components/custom/header.tsx [hunk 2]
```diff
@@ -12,11 +12,45 @@ import { LogoutIcon } from "./icons";
...
+    }
+
+    setPrevScroll(currentScroll);
+  };
+
+  useEffect(() => {
+    
+    window.addEventListener("scroll", handleScroll);
+
+    // Cleanup event listener on component unmount to prevent any memory leak
```
Comment (realChakrawarti) [inline src/components/custom/header.tsx:41]: I like this 👍🏽. Thanks for a adding a comment.

**Comment by realChakrawarti** [comment]
@OMIIIEE Thank you for raising the PR. +1 for preview video. 

Could you please address the above comments. Otherwise looks good to me.

File: .env.example
[the remark this answers is not in the document]
Comment (OMIIIEE) [reply .env.example:1]: yes .

File: tsconfig.json (hunk 1 above)
Comment (OMIIIEE) [reply tsconfig.json]: modified . please look through it

**Review by realChakrawarti — APPROVED** [review]

**Pull request merged**